src/validation.cpp

```diff
@@ -2423,34 +2423,43 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
         return true;
     }
 
-    bool fScriptChecks = true;
-    if (!m_chainman.AssumedValidBlock().IsNull()) {
+    const char* script_check_reason;
```

Comment:
nit: Would be better to initialise to `nullptr`, IMO. Sanitizers should catch and warn for initialisation being missed in tested code paths though.